walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,86 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Comparator;
+import java.util.Set;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    // Edge Case no attendees 
+    if (events.isEmpty() && request.getAttendees().isEmpty()) {
+      return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+
+    // Edge Case meeting lasts longer than single day
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+      return Arrays.asList();
+    }
+
+    List<TimeRange> sortedEventTimes = getSortedEventTimes(events, request);
+    return findOpenRanges(sortedEventTimes, request);
+
   }
+
+  private List<TimeRange> getSortedEventTimes(Collection<Event> events, MeetingRequest request) {
```

Comment:
please specific which order here

Reply:
ahh ok so would `getEventsSortedByStartTime` be good, or would a comment be better?

Reply:
getEventsSortedByStartTime sounds good. Ideally all public method and complicated util function should have comment